include/seqan3/range/container/dynamic_bitset.hpp

```diff
@@ -0,0 +1,1479 @@
+     * \tparam other_range_t The type of range to be inserted; must satisfy std::ranges::InputRange and `value_type`
+     *                       must be constructible from reference_t<other_range_t>.
+     * \param[in]      range The sequences to construct/assign from.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of `range`.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    template <std::ranges::InputRange other_range_t>
+    //!\cond
+        requires std::Constructible<value_type, /*ranges::range_reference_t*/reference_t<other_range_t>>
+    //!\endcond
+    constexpr void assign(other_range_t && range) noexcept
+    {
+        assign(std::ranges::begin(range), std::ranges::end(range));
+    }
+
+    /*!\brief Assign from pair of iterators.
+     * \tparam begin_it_type Must satisfy std::ForwardIterator and the `value_type` must be constructible from
+     *                       the reference type of begin_it_type.
+     * \tparam   end_it_type Must satisfy std::Sentinel.
+     * \param[in]   begin_it Begin of range to construct/assign from.
+     * \param[in]     end_it End of range to construct/assign from.
+     *
+     * ### Complexity
+     *
+     * Linear in the distance between `begin_it` and `end_it`.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    template <std::ForwardIterator begin_it_type, std::Sentinel<begin_it_type> end_it_type>
+    constexpr void assign(begin_it_type begin_it, end_it_type end_it) noexcept
+    //!\cond
+        requires std::Constructible<value_type, /*ranges::iter_reference_t*/reference_t<begin_it_type>>
+    //!\endcond
+    {
+        clear();
+        insert(cbegin(), begin_it, end_it);
+    }
+    //!\}
+
+    /*!\name Iterators
+     * \{
+     */
+    //!\brief Returns the begin to the dynamic_bitset.
+    constexpr iterator begin() noexcept
+    {
+        return iterator{*this};
+    }
+
+    //!\copydoc begin()
+    constexpr const_iterator begin() const noexcept
+    {
+        return const_iterator{*this};
+    }
+
+    //!\copydoc begin()
+    constexpr const_iterator cbegin() const noexcept
+    {
+        return const_iterator{*this};
+    }
+
+    //!\brief Returns iterator past the end of the dynamic_bitset.
+    constexpr iterator end() noexcept
+    {
+        return iterator{*this, size()};
+    }
+
+    //!\copydoc end()
+    constexpr const_iterator end() const noexcept
+    {
+        return const_iterator{*this, size()};
+    }
+
+    //!\copydoc end()
+    constexpr const_iterator cend() const noexcept
+    {
+        return const_iterator{*this, size()};
+    }
+    //!\}
+
+    /*!\name Bit manipulation
+     * \{
+     */
+    //!\brief Sets the bits to the result of binary AND on corresponding pairs of bits of *this and rhs.
+    constexpr dynamic_bitset & operator&=(dynamic_bitset const & rhs) noexcept
+    {
+        assert(size() == rhs.size());
+        data_ &= rhs.data_;
```

Comment:
ah okay both have the same capacity AND same size, so the first bits (for the size of bits) stay the same.